components/customer_io/customer_io.app.mjs

```diff
@@ -1,11 +1,62 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "customer_io",
-  propDefinitions: {},
+  propDefinitions: {
+    customerId: {
+      type: "string",
+      label: "Customer ID",
+      description: "The unique identifier for the customer.",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getBaseUrl() {
+      return "https://track.customer.io/api/v1";
+    },
+    _getHeaders() {
+      const basicauthUserPwd = `${this.$auth.site_id}:${this.$auth.api_key}`;
```

Comment:
Move `this.$auth.api_key` and `this.$auth.site_id` to a getter method.

```suggestion
      const basicauthUserPwd = `${this._siteId()}:${this._apiKey()}`;
```

Reply:
Not possible, nothing is returned